core/mutator/entry/mutation.go

```diff
@@ -148,3 +153,19 @@ func (m *Mutation) sign(signers []signatures.Signer) (*pb.Entry, error) {
 	m.entry.Signatures = sigs
 	return m.entry, nil
 }
+
+// Check verifies that an update was successfully applied.
+// Returns nil if newLeaf is equal to the entry in this mutation.
+func (m *Mutation) Check(newLeaf []byte) error {
+	// XXX Mutations are no longer stable serialized byte slices, so we need to use
+	// an equality operation on the proto itself.
+	leafValue, err := FromLeafValue(newLeaf)
+	if err != nil {
+		return fmt.Errorf("failed to decode current entry: %v", err)
+	}
+
+	if got, want := leafValue, m.entry; !proto.Equal(got, want) {
+		return errors.New("newLeaf does not match expected value")
```

Comment:
I've seen advice that there's no special reason to use errors.New just because you have no varargs items for a fmt.Errorf() call.  But I think you have pointed me to https://en.wikipedia.org/wiki/Uncontrolled_format_string on this topic, am I recalling correctly?

Reply:
That's right.  A secure way of using `fmt.Errorf` in this case would look like:
```go
 fmt.Errorf("%v", "A string with no variables")
```